people-counting-system/people-counting-client/src/app/device-counts/device-counts.page.ts

```diff
@@ -69,38 +77,104 @@ import { addIcons } from 'ionicons';
     IonSelectOption
   ],
 })
+
 export class DeviceCountsPage implements OnInit {
-  startDate: string = new Date().toISOString();
-  endDate: string = new Date().toISOString();
-  deviceId: number | undefined;
+  startDate: string = new Date(new Date().setHours(0, 0, 0, 0)).toISOString(); // beginning of today
+  endDate: string = new Date().toISOString(); // now
+  deviceId: number
   deviceName: string | undefined;
-
-  aggregate: 'minute' | 'hour' | 'day' | 'week' = 'minute';
+  aggregate: CountAggregate = 'hour'; // default aggregation
+  device$!: Observable<Device>;
 
   showStartPicker = false;
   showEndPicker = false;
 
   // Sample chart options, needs replaced later with actual data
-chartOptions = {
-
-};
-  constructor(private route: ActivatedRoute, private router: Router) { 
-    addIcons({
-      refreshOutline,
-      arrowBack
-    });
-  }
+  chartOptions: any = {};
+  errorMsg: string | null = null;
+
+  constructor(  
+    private route: ActivatedRoute,  
+    private countService: CountService,  
+    private deviceService: DeviceService,  
+    private router: Router  
+  ) {  
+    addIcons({  
+      refreshOutline,  
+      arrowBack,  
+    });  
 
-  onRefresh() {
-    console.log('Refreshing chart with:');
-    console.log('Start Date:', this.startDate);
-    console.log('End Date:', this.endDate);
-    console.log('Aggregate:', this.aggregate);
-  }
-  ngOnInit() {
     this.deviceId = Number(this.route.snapshot.paramMap.get('id'));
-    console.log('Loaded counts for device ID:', this.deviceId);
+    if (isNaN(this.deviceId)) {
+    this.router.navigate(['/device']);
+    }  
+  }
 
-    this.deviceName = `Device #${this.deviceId}`;  //This will need changed when service is implemented
+onRefresh() {  
+    this.countService  
+      .getCountsByDevice(  
+        this.deviceId,  
+        this.startDate,  
+        this.endDate,  
+        this.aggregate  
+      )  
+      .pipe(  
+        map((res) => this.mapCountsToChart(res)),  
+        catchError((error) => {  
+          console.error(error);  
+          return of({});  
+        })  
+      )  
+      .subscribe((chartOptions) => {  
+        this.chartOptions = chartOptions;  
+      });  
+  }  
```

Comment:
Set the chart options in a tap instead
```suggestion
onRefresh() {
    this.countService
      .getCountsByDevice(
        this.deviceId,
        this.startDate,
        this.endDate,
        this.aggregate
      )
      .pipe(
        map((res) => this.mapCountsToChart(res)),
        tap((chartOptions) => {
          this.chartOptions = chartOptions;
        }),
        catchError((error) => {
          console.error(error);
          return of({});
        })
      )
      .subscribe();
  }  
```